Makefile

```diff
@@ -130,11 +135,21 @@ asup:
 	@rm -rf autosupport/asup
 	@rm -rf ${ASUP_TMP}
 	@mkdir ${ASUP_TMP}
-	@git clone https://${GIT_TOKEN}@github.com/NetApp/harvest-private.git ${ASUP_TMP}
+	@if [[ $(shell git ls-remote --heads  https://${GIT_TOKEN}@github.com/NetApp/harvest-private.git ${BRANCH} | wc -l | xargs) == 0 ]]; then\
+		git clone -b main https://${GIT_TOKEN}@github.com/NetApp/harvest-private.git ${ASUP_TMP};\
+	else\
+		git clone -b ${BRANCH} https://${GIT_TOKEN}@github.com/NetApp/harvest-private.git ${ASUP_TMP};\
+	fi
```

Comment:
build against asup main branch if harvest-private doesn't have same branch name.

Reply:
good catch - probably worth including your comment in the script since it makes it clearer what's going on

Reply:
done